### PR TITLE
APM-4463: migration to .NET6

### PR DESCRIPTION
**Pull request by olga-tumanova**
**JIRA ticket**
https://checkout.atlassian.net/browse/APM-4463

**Description**
iDEAL migration to .NET 6 caused the exceptions in iDealAdvancedConnector during create payments. 
The issue is known [signed XML CheckSignature always return Null Reference exception after upgrading to .Net 5 · Issue #55383 · dotnet/runtime (github.com)](https://github.com/dotnet/runtime/issues/55383)
The reason is that sha256 algorithm is built in starting from .NET5 and adding it manually leads to runtime exceptions. Because of it we needed to update iDealAdvancedConnector to .NET6.

**Tests**
Using console in solution and test settings that are used in iDEAL
image

### CODE REVIEW

**Comment by andre-lopes-cko** [comment]
Could you add a description of what these changes mean?

**Comment by olga-tumanova** [comment]
> Could you add a description of what these changes mean?

Description is added

File: src/iDealAdvancedConnector/XmlSignature/XmlSignature.cs [hunk 1]
```diff
@@ -22,13 +22,13 @@ public class XmlSignature
...
         public const string XmlDigSignRSASHA256Namespace = "http://www.w3.org/2001/04/xmldsig-more#rsa-sha256";
 
-        /// <summary>
-        /// Registers the signing algorithm
-        /// </summary>
-        public static void RegisterSignatureAlghorighm()
-        {
-            CryptoConfig.AddAlgorithm(typeof(RSAPKCS1SHA256SignatureDescription), XmlDigSignRSASHA256Namespace);
-        }
+        ///// <summary>
```
Comment (andre-lopes-cko) [inline src/iDealAdvancedConnector/XmlSignature/XmlSignature.cs]: If it is not needed, we need to remove it.

**Comment by andre-lopes-cko** [comment]
I think there are some tests there. did you run them?
I think in the Console app

**Comment by olga-tumanova** [comment]
> I think there are some tests there. did you run them? I think in the Console app

They were not working before my changes. And fail with the same error after my changes

File: src/iDealAdvancedConnector/XmlSignature/XmlSignature.cs (hunk 1 above)
Comment (olga-tumanova) [reply src/iDealAdvancedConnector/XmlSignature/XmlSignature.cs]: Removed

**Review by andre-lopes-cko — APPROVED** [review]

**Pull request merged**